data-prepper-plugins/translate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/translate/MappingsParameterConfig.java

```diff
@@ -54,11 +47,29 @@ public boolean isSourceFieldValid() {
         }
         if (source instanceof List<?>) {
             List<?> sourceList = (List<?>) source;
-            return sourceList.stream().allMatch(sourceItem -> sourceItem instanceof String);
+            if(sourceList.isEmpty()){
+                return false;
+            }
+            return sourceList.stream().allMatch(sourceItem -> sourceItem instanceof String)
+                   && commonRootPath(sourceList);
```

Comment:
Is there a reason all the `source` keys have to have a common root path? If not we should allow different source keys. Can be in future PR